few edits (#9)
stemangiola/bioc_2020_tidytranscriptomics: vignettes/tidytranscriptomics.Rmd

# Filtering lowly expressed genes  
Genes with very low counts across all libraries provide little evidence for differential expression and they interfere with some of the statistical approximations that are used later in the pipeline. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes should be filtered out prior to further analysis.

> With tidybulk, it is not really needed to explicitly filter lowly transcribed genes as all calulations (e.g., scaling, removal unwanted variation, differential expression testing) are done on abundantly transcribed genes; although in case of scaling, it is apllied back on all transcripts.

Reviewer: ```suggestion
With tidybulk, it is not really necessary to explicitly filter lowly transcribed genes, as all calculations (e.g., scaling, removal of unwanted variation, differential expression testing) are performed on abundantly transcribed genes; although in case of scaling, the scaling is applied back to all genes in the dataset.
```

However, to explicitly and/or temporarly drop them from the data set is useful for data visualisation. Tidybulk, uses the edgeR filterByExpr function described [here](https://f1000research.com/articles/5-1408). By default, this will keep genes with ~10 counts in a minimum number of samples, the number of the samples in the smallest group. In this dataset the smallest group size is 2 samples. tidybulk performs this filtering in the functions we will use `scale_abundance()` and `test_differential_abundance()` and we can explicitely apply it with `keep_abundant()`.

```{r}

> # Take a look to only highly abundant genes 

Reviewer: ```suggestion
# Take a look at the abundant genes
```

After we run `scale_abundance()` we should see some columns have been added at the end. We have a column called `lowly_abundant` that indicates whether the gene has been filtered due to being lowly expressed. FALSE means the gene wasn’t filtered, TRUE means it was. The `count_scaled` column contains the scaled counts.

> We can now see the difference of abundance densities before and adfter scaling

Reviewer: ```suggestion
We can now see the difference of abundance densities before and after scaling
```

counts_scal_MDS %>% 
	
	# extract 500 most variable genes
	keep_variable( .abundance = count_scaled, top = 500) %>%
		
	# create heatmap
	heatmap(
	      .column = sample,
	      .row = transcript,
	      .value = count_scaled,
	      annotation = c(dex, cell),

> 	      transform = function(x) x %>% log1p %>% tidyHeatmap:::scale_robust()

Reviewer: this is a failure here:

```Quitting from lines 390-403 (tidytranscriptomics.Rmd) 
Error: processing vignette 'tidytranscriptomics.Rmd' failed with diagnostics:
object 'scale_robust' not found
--- failed re-building ‘tidytranscriptomics.Rmd’

is scale_robust only available in tidybulk dev? we're supposed to be able to use bioc devel so is there a way to get it into that?

Author: I will make it exported from tidyHeatmap and update CRAN. Strange that gives error to you. Aa maybe because you have the cran version of course. I will update CRAN don't worry

Author: In the meanwhile could you please update tidyHeatmap from github?

Reviewer: I could but that won't stop the failure, because Github actions is automatically installing the packages from CRAN and Bioc devel, you can see the details below if you haven't already
<img width="925" alt="Screen Shot 2020-05-30 at 7 17 50 pm" src="https://user-images.githubusercontent.com/5591821/83324736-b7340080-a2aa-11ea-805c-40cf27ca3043.png">

<img width="1419" alt="Screen Shot 2020-05-30 at 7 17 21 pm" src="https://user-images.githubusercontent.com/5591821/83324754-c6b34980-a2aa-11ea-964d-f3d52a20ad8d.png">

If it fails it means the changes don't get pushed to our website that the Github actions are automatically building here https://stemangiola.github.io/bioc_2020_tidytranscriptomics/articles/tidytranscriptomics.html

So if the change will get to CRAN soon we could wait otherwise if it will take a while (more than a day?) it's probably better to remove that scale_robust so we can add the other changes, then add the scale_robust when it's in CRAN

Author: I see. The CRAN tidyHeatmap it does scale automatically. I see we were seing to different plots because you have the CRAN version. I as seing it not scaled and so very ugly.

In this case discart my change!

Reviewer: ok will do 😄

> 	      transform = function(x) x %>% log1p %>% tidyHeatmap:::scale_robust()

Reviewer: ```suggestion
	      transform = log1p
```
So I think if you accept this suggested change to revert this PR should hopefully pass
